pkg/distributor/ha_tracker.go

```diff
@@ -402,6 +415,34 @@ func (h *haTracker) cleanupOldReplicas(ctx context.Context, deadline time.Time)
 	}
 }
 
+// HaShortcutRequestChecker will check whether the sample in the request must be parsed
+// in the next step or can be shortcut avoiding unneeded request parsing.
+func (h *haTracker) HaShortcutRequestChecker() func(ctx context.Context, w http.ResponseWriter, r *http.Request) bool {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) (finished bool) {
```

Comment:
Why do we need to return a function here at all? I don't think we need to, this can be:
```suggestion
func (h *haTracker) ShouldShortcutRequest(ctx context.Context, w http.ResponseWriter, r *http.Request) (finished bool) {
```

Some nits:
- I removed `HA` because this is the HA tracker, it only does HA tracking things.
- I renamed function to a question, as it returns a bool.